### DISCUSSION TITLE
Operations vs LazyOperations vs Commands?

### DISCUSSION THREAD

**Discussion by damccull** [ravendb/ravendb#15526] [Q&A]
To begin with, what is the difference between an Operation and a Command?

I'm still working to get my head around the source (jvm version) and this is something I haven't figured out yet. Here's what I think, though: Commands are individual units sent to the server either individually or as part of an operation to cause a specific, single result. Operations, on the other hand, are collections of commands in a specific order designed to create an overall effect on the database through the sequenced application of units of work. Is this correct?

Second question: What's the difference between an Operation and a LazyOperation?

Is it as simple as a LazyOperation only being stored clientside until a batch of them are sent together, while an Operation is sent immediately?

**Comment by ayende** [comment]
Operation is 1:1 with command, most often.
Command is about sending the request to the server and processing the result.
Operation is about integrating that result to the client side. For example, see `LoadOperation` vs `GetDocumentsCommand`.

The `LoadOperation` is responsible for integrating with the unit of work, etc.

Lazy operation is about batching multiple calls to the server into a single roundtrip.
